Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/repository/FamilyTestRepository.kt

```diff
@@ -57,52 +58,38 @@ class FamilyTestRepository @Inject constructor(
         }
     }
 
-    suspend fun refresh(): Map<TestIdentifier, Exception> {
-        val exceptions = mutableMapOf<TestIdentifier, Exception>()
-        familyTests.first().filterNot {
+    suspend fun refresh(): Set<Error> = coroutineScope {
```

Comment:
This is dangerous because we are losing thread-safety. While the server is polling the tests, the test could be changed and updated in the database by other operations. Then when the polling is finished, the tests still have the old, cached state and will override the state in the database..

Reply:
what is exactly you see dangerous? parallel loading ? or ?

Reply:
I reverted all, I kept the navigation fixes